components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -1421,4 +1429,26 @@ private void setLoggedInOrgIdInRequest(AuthenticatedIdPData authenticatedIdPData
             }
         }
     }
+
+    /**
+     * Resolve user resident organization for the users authenticated via the B2B organization login authenticator.
+     *
+     * @param authenticatedUser   The authenticated user.
+     * @param authenticatorConfig The authenticator configurations.
+     * @return The organization where the user's identity is managed.
+     */
+    private String resolveUserResidentOrganization(AuthenticatedUser authenticatedUser,
+                                                   AuthenticatorConfig authenticatorConfig) throws FrameworkException {
+
+        // Check for user organization claim for the authenticated user via the organization login authenticator.
+        if (authenticatedUser.getUserAttributes() != null && isLoggedInWithOrganizationLogin(authenticatorConfig)) {
```

Comment:
You already checked whether user authenticated with org login
```suggestion
        if (authenticatedUser.getUserAttributes() != null) {
```

Reply:
Fixed by - https://github.com/wso2/carbon-identity-framework/pull/4915/commits/ae978f284202c078d3e24494b68339e7c653bae6